Buildings/Fluid/SolarCollectors/Data/BaseClasses/Generic.mo

```diff
@@ -0,0 +1,88 @@
+within Buildings.Fluid.SolarCollectors.Data.BaseClasses;
+record Generic
+  "Generic data record providing common inputs for ASHRAE93 and EN12975 solar collector models"
+  extends Modelica.Icons.Record;
+
+  parameter Modelica.Units.SI.Area A "Area";
+  parameter Buildings.Fluid.SolarCollectors.Types.HeatCapacity CTyp
+    "Total thermal capacity or fluid volume and 'dry' thermal capacity or mass";
+  parameter Modelica.Units.SI.HeatCapacity C
+    "Dry or total thermal capacity of the solar thermal collector";
+  parameter Modelica.Units.SI.Volume V "Fluid volume ";
+  parameter Modelica.Units.SI.Mass mDry "Dry mass";
+  parameter Real mperA_flow_nominal(final unit="kg/(s.m2)")
+    "Nominal mass flow rate per unit area of collector";
+  parameter Modelica.Units.SI.PressureDifference dp_nominal(displayUnit="Pa")
+    "Nominal pressure drop";
+  parameter Modelica.Units.NonSI.Angle_deg[:] incAngDatDeg={0,10,20,30,40,50,60,70,80,90}
+    "Incident angle data in degrees";
```

Comment:
This would be the first occurrence were we use degree in the library. I think this can be avoided by using rad and set the `displayUnit`.

Reply:
If the users define their own characteristics, they'll do this in text mode and there the `displayUnit` has no effect (for `Modelica.Units.SI.Angle` the default `displayUnit` is degrees).
 
Maybe we can remove `incAngDatDeg` and define `incAngDat` as `Modelica.Units.Conversions.from_deg({0,10,20,30,40,50,60,70,80,90})`?

With this change we also need to update the function `validateAngles`.

Let me know what you think and I can make the necessary changes.

Reply:
@jelgerjansen : It would be good if you can remove `incAngDatDeg` and replace it with `incAngDat` and call the conversion function.

Reply:
@mwetter I pushed the required changes to [my repo](https://github.com/jelgerjansen/modelica-buildings/tree/issue3604_solarCollectors) (commit 62acd6631d).

Reply:
Thanks, I merge them.